nomad/structs/structs.go

```diff
@@ -12215,6 +12217,95 @@ type ACLTokenUpsertResponse struct {
 	WriteMeta
 }
 
+type ACLAuthMethod struct {
+	Name          string
+	Type          string
+	TokenLocality string
+	MaxTokenTTL   string
+	Default       bool
+	Config        *ACLAuthMethodConfig
+
+	Hash []byte
+
+	CreateTime  time.Time
+	ModifyTime  time.Time
+	CreateIndex uint64
+	ModifyIndex uint64
+}
+
+// SetHash is used to compute and set the hash of the ACL auth method. This
+// should be called every and each time a user specified field on the method is
+// changed before updating the Nomad state store.
```

Comment:
Right—I'm not sure. My reasoning was that the only possibly user-definable field should be the `Name`, because all other properties would be dictated by the token provider? But I think you're right that basically all the other fields could possibly be user-definable.